apps/test/unit/javalab/JavalabEditorTest.js

```diff
@@ -56,31 +50,33 @@ describe('Java Lab Editor Test', () => {
       expect(submitBtn.prop('value')).to.equal('Save');
     });
 
-    it('calls JavalabFileManagement functions and updates state on Rename save', () => {
+    it('updates state on Rename save', () => {
       const editor = createWrapper();
 
+      // should have default file in redux
+      expect(store.getState().javalab.sources['MyClass.java']).to.not.be
+        .undefined;
+
       const activateRenameBtn = editor.find('button').first();
       activateRenameBtn.invoke('onClick')();
 
       // first input should be file rename text input
       const renameInput = editor.find('input').first();
       renameInput.invoke('onChange')({target: {value: 'NewFilename.java'}});
 
-      // save button not clicked, should not yet have set project changed
-      // or change filename in redux
-      expect(JavalabFileManagement.onProjectChanged).to.not.have.been.called;
-      expect(store.getState().javalab.filename).to.not.equal(
-        'NewFilename.java'
-      );
+      // save button not clicked, should not yet have changed filename in redux
+      expect(store.getState().javalab.sources['NewFilename.java']).to.be
```

Comment:
NIT: Could you stick these filenames in variables (i.e. originalName, newName or similar). I was having trouble following along with which parts were changing.

Reply:
done!